kartothek/io/eager.py

```diff
@@ -738,3 +742,60 @@ def garbage_collect_dataset(dataset_uuid=None, store=None, factory=None):
     # Given that `nested_files` is a generator with a single element, just
     # return the output of `delete_files` on that element.
     return delete_files(next(nested_files), store_factory=ds_factory.store_factory)
+
+
+def copy_dataset(
+    src_dataset_uuid: str,
+    store: KeyValueStore,
+    target_dataset_uuid: Optional[str] = None,
+    tgt_store: Optional[KeyValueStore] = None,
+):
+    """
+    Copies and optionally renames a dataset, either  from one store to another or
+    within one store.
+
+    :param src_dataset_uuid:
+        UUID of source dataset
+    :param store:
+        Source store
+    :param target_dataset_uuid:
+        UUID of target dataset. May be the same as src_dataset_uuid, if store
+        and tgt_store are different. If empty, src_dataset_uuid is used
+    :param tgt_store:
+        Target Store. May be the same as store, if src_dataset_uuid and
+        target_dataset_uuid are different. If empty, value from parameter store is
+        used
+    """
+    if target_dataset_uuid is None:
+        target_dataset_uuid = src_dataset_uuid
+    if tgt_store is None:
+        tgt_store = store
+
+    if (src_dataset_uuid == target_dataset_uuid) & (store == tgt_store):
```

Comment:
Shouldn't this just be a no op rather than a ValueError?

Reply:
Both are fine for me. However, in similar situations the KTK API raises a ValueError, e.g. https://github.com/JDASoftwareGroup/kartothek/blob/08a809429cba9f46c4c0d09853eb476021764708/kartothek/io/eager.py#L362.
So leaving the error here as is would be at least consistent with the rest of the API.